### PR TITLE
Use CSS grid for caption Setting overlay to avoid cutoff

### PR DESCRIPTION
**Pull request by RevinKey**
## Description
Currently the Caption settings menu has two issues:
1. extra characters added to the classname here: https://github.com/videojs/video.js/blob/master/src/js/tracks/text-track-settings.js#L437
2. When the player is smaller than ~450px the menu is squished together and some items are cut off. 

## Enhancements/ Changes
1. Removed the extra characters from the class name so the CSS will target correctly
2. Switch to CSS Grid with minmax column row size to make sure the menu is responsive and looks uniform at all sizes. 
**Before:**
Large Player:
hiddenbuttons

Small Player: 
screenshot2

**After:**
Large Player:
screen shot 2017-12-21 at 12 56 48 pm

Small Player: 
screen shot 2017-12-21 at 12 56 39 pm

### CODE REVIEW

**Comment by RevinKey** [comment]
This is a new PR for #4835 since I somehow broke my branch

**Comment by gkatsev** [comment]
Just tried this out on the live test page: https://deploy-preview-4997--videojs-docs.netlify.com/test-example/
and the right column isn't aligned correctly.

File: src/css/components/_captions-settings.scss [hunk 1]
```diff
@@ -21,6 +21,48 @@
   vertical-align: bottom;
 }
 
+// code that will only run if CSS Grid is supported by the browser
+@supports (display: grid) {
+  .vjs-text-track-settings .vjs-modal-dialog-content {
+    display: grid;
+    grid-template-columns: 1fr auto;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: the second auto should be `1fr` here

File: src/css/components/_captions-settings.scss [hunk 2]
```diff
@@ -21,6 +21,48 @@
...
+
+  .vjs-text-track-settings .vjs-track-settings-font {
+    grid-column: 2;
+    grid-row: 1;
+  }
+
+  .vjs-text-track-settings .vjs-track-settings-controls {
+    grid-column: 2;
+    grid-row: 2;
+    text-align: unset;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: why are these two unset?

File: src/css/components/_captions-settings.scss [hunk 3]
```diff
@@ -35,7 +77,7 @@
 
 .vjs-text-track-settings legend {
   color: $primary-foreground-color;
-  margin: 0 0 5px 0;
+  margin: 0 0 5px;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: can you revert this change and the ones below it? It'll keep the PR focused on just necessary code changes.

File: src/css/components/_captions-settings.scss (hunk 2 above)
Comment (gkatsev) [reply src/css/components/_captions-settings.scss]: not unsetting them keeps the buttons visually at the bottom right where they should be.

File: src/css/components/_captions-settings.scss [hunk 4]
```diff
@@ -21,6 +21,48 @@
...
+    grid-row: 2;
+    text-align: unset;
+    vertical-align: unset;
+  }
+  .vjs-text-track-settings fieldset span.vjs-opacity {
+    margin-left: unset;
+  }
+  .video-js {
+    @media (max-width: 365px) {
+      .vjs-modal-dialog .vjs-modal-dialog-content {
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: we shouldn't set generic modal-dialog properties inside this file. We should target the text-track-settings specifically.

Comment (gkatsev) [reply src/css/components/_captions-settings.scss]: Also, this doesn't seem to be getting applied?

File: src/css/components/_captions-settings.scss (hunk 3 above)
Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: Yup my code beautifier messed up a few things like these. Working right now to get it back to how it should be

File: src/css/components/_captions-settings.scss (hunk 4 above)
Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: I had this in there to resize the font if the player was a certain size. but your are correct doesn't seem to work, nor make a beneficial difference in appearance when I just tested manually

File: src/css/components/_captions-settings.scss (hunk 2 above)
Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: I thought it gave it a weird 3 column effect. But i can revert this if you'd like.

Comment (gkatsev) [reply src/css/components/_captions-settings.scss]: yeah, I think I'd prefer to keep it as is, we can change it up a bit later on if that was wanted.

File: src/css/components/_captions-settings.scss [hunk 5]
```diff
@@ -21,6 +21,36 @@
...
+  }
+
+  .vjs-text-track-settings .vjs-track-settings-controls {
+    grid-column: 2;
+    grid-row: 2;
+    text-align: unset;
+    vertical-align: unset;
+  }
+  .vjs-text-track-settings fieldset span.vjs-opacity {
+    margin-left: unset;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: unsetting the margin makes the dropdowns too close. Should probably be removed too.

Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: just updated it to give the margin-right on the <select> and keep the left margin unset. this way it will have a gap between them still and be aligned vertically on small players

File: src/css/components/_captions-settings.scss [hunk 6]
```diff
@@ -21,6 +21,40 @@
...
+  }
+
+  .vjs-text-track-settings .vjs-track-settings-controls {
+    grid-column: 2;
+    grid-row: 2;
+
+  }
+
+  .vjs-text-track-settings fieldset span.vjs-opacity {
+    margin-left: unset;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: instead of having this and the margin-right, should we just update the default styling below to have a margin-right?

Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: Just tried it out and it works good in both the grid and table moving this to the default style.

File: src/css/components/_captions-settings.scss [hunk 7]
```diff
@@ -30,7 +60,7 @@
 
 .vjs-text-track-settings fieldset span {
   display: inline-block;
-  margin-left: 5px;
+  margin-left: 0px;
```
Comment (gkatsev) [inline src/css/components/_captions-settings.scss]: I think this line is unnecessary now.

Comment (RevinKey) [reply src/css/components/_captions-settings.scss]: yes you are correct, just pushed to remove this.

**Comment by bcbclifford** [comment]
This is a step in the right direction but more is needed for smaller players. Items are still cropped at ~450px for a 16:9 player. We could reduce padding, switch to a single column and show the scrollbar.

**Comment by gkatsev** [comment]
@bcbclifford those are good changes, I'm not sure we want to make them as part of this PR though.
